File: app/src/main/java/com/bas/google_book_app/network_service/GoogleBookAPI.java
package com.bas.google_book_app.network_service;

import com.bas.google_book_app.domain.dto.BookResponseDTO;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface GoogleBookAPI {

    @GET("volumes/")
    Call<BookResponseDTO> getBooks(
            @Query("key") String apiKey,
            @Query("q") String query,
            @Query("startIndex") int page,
            @Query("maxResults") int maxResults,
            @Query("projection") String projection
    );
}

File: app/src/main/java/com/bas/google_book_app/network_service/GoogleBookAPI.kt
package com.bas.google_book_app.network_service

import com.bas.google_book_app.domain.dto.BookResponseDTO
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface GoogleBookAPI {
    @GET("volumes/")
    open fun getBooks(
        @Query("key") apiKey: String?,
        @Query("q") query: String?,
        @Query("startIndex") page: Int,
        @Query("maxResults") maxResults: Int,
        @Query("projection") projection: String?
    ): Call<BookResponseDTO?>?
}